pyproject.toml

```diff
@@ -388,12 +389,18 @@ lint.ignore = [  # NOTE: non-permanent exclusions should be added to `.ruff.toml
     "B018",  # UselessExpression
     "D",  # pydocstyle
     "PGH001",  # No builtin eval() allowed
+]
+"**/tests/*.py" = [
+    "S101",  # Use of assert detected
+]
+"astropy/nddata/_testing.py" = [
     "S101",  # Use of assert detected
 ]
 ".pyinstaller/*.py" = ["INP001"]  # Not a package.
 "conftest.py" = ["INP001"]  # Part of configuration, not a package.
 "docs/*.py" = [
     "INP001",  # implicit-namespace-package. The examples are not a package.
+    "S101",  # Use of assert detected
```

Comment:
It is not at all clear to me why these edits to the permanent repository-level Ruff configuration are being made.

Reply:
`assert`s are always going to be needed for pytest but can't be used in runtime code if `-OO` is going to be supported. This seems like a permanent thing rather than temporary ignore to me, but if it's better suited for `ruff.toml` I'll move it.

Reply:
I guess this gets a bit to a larger question: the few `assert` I inserted were really for things that should not happen (i.e., just there so that if someone trips it, we get to know it). It is in fact OK if those are skipped with `-O0`. That said, I think it is also fine to annotate those with an exception to the `ruff` checks, since really this should be unusual.

Reply:
Agreed that the assert statements in `docs/wcs/examples/from_file.py` are instructive in a nice, concise way, that would be less good if `# noqa` entries were added. They are also good examples of things that should be skipped with `-OO`.

I'm personally fine with either explicitly removing the two relevant files or just removing everything in `docs` - but in any case, let's just make a decision and do it - the cost/benefit ratio of many of these ruff style nitpicks are marginal as it is...